src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/aws/AwsConnector.java

```diff
@@ -175,31 +138,40 @@ public void buildStack(Stack stack, String userData, Map<String, Object> setupPr
         MDCBuilder.buildMdcContext(stack);
         AwsCredential awsCredential = (AwsCredential) stack.getCredential();
         AmazonCloudFormationClient client = awsStackUtil.createCloudFormationClient(Regions.valueOf(stack.getRegion()), awsCredential);
-        String stackName = String.format("%s-%s", stack.getName(), stack.getId());
-
+        String cFStackName = String.format("%s-%s", stack.getName(), stack.getId());
         List<Parameter> parameters = new ArrayList<>(Arrays.asList(
                 new Parameter().withParameterKey("SSHLocation").withParameterValue("0.0.0.0/0"),
                 new Parameter().withParameterKey("CBUserData").withParameterValue(userData),
-                new Parameter().withParameterKey("StackName").withParameterValue(stackName),
+                new Parameter().withParameterKey("StackName").withParameterValue(cFStackName),
                 new Parameter().withParameterKey("StackOwner").withParameterValue(awsCredential.getRoleArn()),
                 new Parameter().withParameterKey("KeyName").withParameterValue(awsCredential.getKeyPairName()),
                 new Parameter().withParameterKey("AMI").withParameterValue(stack.getImage()),
                 new Parameter().withParameterKey("RootDeviceName").withParameterValue(getRootDeviceName(stack, awsCredential))
         ));
         CreateStackRequest createStackRequest = createStackRequest()
-                .withStackName(stackName)
+                .withStackName(cFStackName)
                 .withOnFailure(com.amazonaws.services.cloudformation.model.OnFailure.valueOf(stack.getOnFailureActionAction().name()))
                 .withTemplateBody(cfTemplateBuilder.build("templates/aws-cf-stack.ftl",
                         spotPriceNeeded(stack.getInstanceGroups()), stack.getInstanceGroupsAsList()))
-                .withNotificationARNs((String) setupProperties.get(SnsTopicManager.NOTIFICATION_TOPIC_ARN_KEY))
                 .withParameters(parameters);
         client.createStack(createStackRequest);
+        Set<Resource> resourceSet = new HashSet<>();
+        Resource resource = new Resource(ResourceType.CLOUDFORMATION_STACK, cFStackName, stack, null);
+        resourceSet.add(resource);
+        stack = stackUpdater.addStackResources(stack.getId(), Arrays.asList(resource));
+        LOGGER.info("CloudFormation stack creation request sent with stack name: '{}' for stack: '{}'", cFStackName, stack.getId());
+
+        List<StackStatus> errorStatuses = Arrays.asList(CREATE_FAILED, ROLLBACK_IN_PROGRESS, ROLLBACK_FAILED);
```

Comment:
CF stacks can get into a ROLLBACK_COMPLETED state (I know that ROLLBACK_IN_PROGRESS comes first but if it is missed for some reason - e.g.: rolled back too fast between 2 polling attempts - it will go on forever)